### PR TITLE
Put the subcmd name in the validate_output error message [#1539]

### PR DESCRIPTION
**Pull request by genehack**
## Description of proposed changes

This puts the full sub-command name into the error message shown when the output records have inconsistent field names.

## Related issue(s)

#1539 

## Checklist

- [x] Automated checks pass
- [x] [Check][1] if you need to add a changelog message
- [x] [Check][2] if you need to add tests
- [x] [Check][3] if you need to update docs

[1]: https://github.com/nextstrain/augur/blob/-/docs/contribute/DEV_DOCS.md#updating-the-changelog
[2]: https://github.com/nextstrain/augur/blob/-/docs/contribute/DEV_DOCS.md#testing
[3]: https://github.com/nextstrain/augur/blob/-/docs/contribute/DEV_DOCS.md#when-to-update

### CODE REVIEW

File: augur/curate/__init__.py
Comment (joverlee521) [inline augur/curate/__init__.py:221]: The subcommand name should be directly accessible because we specify `dest="subcommand"` in 

https://github.com/nextstrain/augur/blob/08e35eec25d99339e23b4298d5685577fee4af67/augur/curate/__init__.py#L112-L113

```suggestion
    subcmd_name = args.subcommand
```

**Review by joverlee521 — COMMENTED** [review]
My comment in https://github.com/nextstrain/augur/pull/1518 came back to bite real quick 😅 

> there's no good way to check for the output error. Hopefully we just never see it!

**Comment by genehack** [comment]
> My comment in #1518 came back to bite real quick 😅
> 
> > there's no good way to check for the output error. Hopefully we just never see it!

I spent like, 30 seconds thinking about mocking something up to cause the error so I could write a test, then settled for manually validating it on the broken version of `parse-genback-location`...

File: augur/curate/__init__.py
Comment (genehack) [reply augur/curate/__init__.py]: ah, thanks, that felt like i was going the long way 'round... force-pushed.

**Review by joverlee521 — APPROVED** [review]

**Comment by joverlee521** [comment]
Once this is merged, I think good to release along with https://github.com/nextstrain/augur/pull/1531 as Augur 25.1.1

(Probably wait till Monday so we don't do a Friday release 🙈)

**Comment by genehack** [comment]
> I spent like, 30 seconds thinking about mocking something up to cause the error so I could write a test, then settled for manually validating it on the broken version of `parse-genback-location`...

In the harsh light of morning, I realized, while functional tests would be a PITA, I could _totally_ trivially unit test the `validate_records()` method, so I added that.

File: tests/io/test_curate_validate_records.py
Comment (joverlee521) [inline tests/io/test_curate_validate_records.py:37]: Pytest has a built in way to of [asserting exceptions](https://docs.pytest.org/en/8.2.x/getting-started.html#assert-that-a-certain-exception-is-raised). 

You can see an example in https://github.com/nextstrain/augur/blob/f9671ac381a6cd64065841776fb874c2a0b88909/tests/io/test_metadata.py#L83-L86

Comment (genehack) [reply tests/io/test_curate_validate_records.py]: ah, right, cool, i always forget about this, will convert over.

**Pull request merged**
